services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImpl.java

```diff
@@ -422,4 +548,41 @@ protected void validateClientVerificationStatus(String clientId) throws NotFound
 			throw new OAuthClientNotVerifiedException("The OAuth client (" + clientId + ") is not verified.");
 		}
 	}
+
+	@WriteTransaction
+	@Override
+	public void revokeToken(OAuthTokenRevocationRequest revocationRequest) {
+		switch (revocationRequest.getToken_type_hint()) {
+			case access_token: // retrieve the refresh token ID from the JWT
+				String refreshTokenId = this.getRefreshTokenId(revocationRequest.getToken());
```

Comment:
please make sure there is a test of the case in which a non-JWT is passes as an access token